src/core/qgswebview.h

```diff
@@ -39,11 +39,11 @@ class CORE_EXPORT QgsWebView : public QWebView
     explicit QgsWebView( QWidget *parent = nullptr )
       : QWebView( parent )
     {
-      const QDesktopWidget desktop;
+      const QScreen *currentScreen = this->screen();
       // Apply zoom factor for HiDPI screens
-      if ( desktop.logicalDpiX() > 96 )
+      if ( currentScreen->logicalDotsPerInchX() > 96 )
       {
-        setZoomFactor( desktop.logicalDpiX() / 96 );
+        setZoomFactor( currentScreen->logicalDotsPerInchX() / 96 );
```

Comment:
Wouldn't be better to call QgsScreenHelper ?

Reply:
It definitely would -- BUT QgsScreenHelper is in GUI, while this class is in core 😭. 

@ptitjano if you look at the changes REMOVED from the layout designer dialog in https://github.com/qgis/QGIS/pull/49774/commits/ebeec926c1b97f5368dfed49c272638b187b56c2 then this is the full logic we need to implement in QgsWebView. Are you able to do that?